### PR TITLE
test analytics: update statement timeout for DML statements

### PR DESCRIPTION
**Pull request by nrainer-materialize**
Observed:
* https://materializeinc.slack.com/archives/C05CQUVHY48/p1721820903825059
* https://materializeinc.slack.com/archives/C05CQUVHY48/p1721828863291669

### CODE REVIEW

File: misc/python/materialize/test_analytics/connector/test_analytics_connector.py [hunk 1]
```diff
@@ -127,7 +127,7 @@ def submit_update_statements(self) -> None:
         if len(self.update_statements) == 0:
             return
 
-        cursor = self.create_cursor(autocommit=not self._use_transaction)
+        cursor = self.create_cursor(autocommit=not self._use_transaction, statement_timeout="2s")
```
Comment (nrainer-materialize) [inline misc/python/materialize/test_analytics/connector/test_analytics_connector.py:130]: Doubled from originally `1s` but we can also go higher... (`5s`?)

File: bazel-mz-repo [hunk 2]
```diff
@@ -0,0 +1 @@
+/private/var/tmp/_bazel_nrainer/d9c0ff8491d02285b22d4a7fd5324111/execroot/_main
```
Comment (def-) [inline bazel-mz-repo:1]: ?

**Review by def- — CHANGES_REQUESTED** [review]

File: misc/python/materialize/test_analytics/connector/test_analytics_connector.py (hunk 1 above)
Comment (def-) [reply misc/python/materialize/test_analytics/connector/test_analytics_connector.py]: Go to 5s, or even 30, I think that's fine.

Comment (def-) [reply misc/python/materialize/test_analytics/connector/test_analytics_connector.py]: Updates are known to have horrible performance, so this is totally expected

**Review by def- — APPROVED** [review]
For reference: https://materialize.com/docs/sql/update/

> Low performance. While processing an UPDATE statement, Materialize cannot process other INSERT, UPDATE, or DELETE statements.

**Pull request merged**